src/SongInSeok/ManagementSystemImpl.java

```diff
@@ -0,0 +1,94 @@
+package SongInSeok;
+
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class ManagementSystemImpl implements ManagementSystem {
+
+    HashMap<Integer, Member> memberGather = new HashMap<>();            //Map 객체 생성
+
+    @Override
+    public boolean addMember(String name, int age, int id) {
+        if (memberGather.containsKey(id)) {return false;}
+
+        memberGather.put(id, new Member(name, age, id));                //id값에 맞는 Member 객체 생성
+        return true;
+    }
+
+
+    @Override
+    public boolean removeMember(int id) {
+        if(memberGather.get(id)==null) {return false;}
+
+        memberGather.remove(id);                                        //키값(id)에 맞는 value 제거
+        return true;
+    }
+
+    @Override
+    public boolean updateMember(int id, String name) {
+        if(memberGather.get(id)==null) {return false;}
+
+        Member modifiedMember = memberGather.get(id);                   //키값(id)에 맞는 객체에 접근 후 객체 변수 값 변경
+        modifiedMember.setName(name);
+        return true;
+    }
+
+    @Override
+    public boolean updateMember(int id, int age) {
+        if(memberGather.get(id)==null) {return false;}
+
+        Member modifiedMember = memberGather.get(id);                   //키값(id)에 맞는 객체에 접근 후 객체 변수 값 변경
+        modifiedMember.setAge(age);
+        return true;
+    }
+
+    @Override
+    public boolean updateMember(int id, String name, int age) {
+        if(memberGather.get(id)==null) {return false;}
+
+        Member modifiedMember = memberGather.get(id);                  //키값(id)에 맞는 객체에 접근 후 객체 변수 값 변경
+        modifiedMember.setName(name);
+        modifiedMember.setAge(age);
+        return true;
+    }
+
+
+    @Override
+    public Member findById(int id) {
+        Member serchMember = memberGather.get(id);                     //키값(id)에 맞는 객체에 접근 후 리턴
+        return serchMember;
+    }
```

Comment:
이렇게 작성하면 불필요한 변수 생성은 없어도 되지 않을까요?

```suggestion
    public Member findById(int id) {
        return memberGather.get(id);                     //키값(id)에 맞는 객체에 접근 후 리턴
    }
```